kubernetes-client/src/main/java/io/fabric8/kubernetes/client/dsl/internal/AbstractWatchManager.java

```diff
@@ -178,6 +185,10 @@ boolean isForceClosed() {
   }
   
   void eventReceived(Watcher.Action action, HasMetadata resource) {
+    if (!receiveBookmarks && action == Action.BOOKMARK) {
```

Comment:
Is that all that's required to support bookmarks? What is needed by watchers to handle bookmark events?

Reply:
If a user has not explicitly set allowWatchBookmarks=true in their ListOptions, then this will filter the events before they reach the Watcher.  Otherwise the built-in handling is already there in AbstractWatchManager.onMessage - we just need updateResourceVersion to be called on a bookmark.

Reply:
It's already there, this is what I was referring to: https://github.com/fabric8io/kubernetes-client/blob/e4d0a9fdc1beef595f922fe4e50aa4e67d709c87/kubernetes-client/src/main/java/io/fabric8/kubernetes/client/dsl/internal/AbstractWatchManager.java#L251